gdx/src/com/badlogic/gdx/scenes/scene2d/ui/Spinner.java

```diff
@@ -0,0 +1,210 @@
+				// Update spinner value, and fire change events if needed
+				if (x >= 0 && x <= getWidth() && y >= 0 && y <= getHeight()) {
+					if (amount > 0) {
+						if (getModel().previous()) Spinner.this.fire(new ChangeListener.ChangeEvent());
+					} else if (amount < 0) {
+						if (getModel().next()) Spinner.this.fire(new ChangeListener.ChangeEvent());
+					}
+					invalidate();
+				}
+				return false;
+			}
+		});
+	}
+
+	protected Drawable getCurrentBackground () {
+		return this.currentBackground;
+	}
+
+	protected void setCurrentBackground (Drawable currentBackground) {
+		this.currentBackground = currentBackground;
+	}
+
+	/** Retrieves the correct background given the location an event occured.
+	 * @param x X coordinate of the fired event
+	 * @param y Y coordinate of the fired event
+	 * @return the background */
+	protected Drawable getBackgroundForClick (float x, float y) {
+		float diffX = getWidth() - x;
+		float diffY = y;
+		if (diffX < getStyle().background.getRightWidth() && diffX >= 0) {
+			if (diffY <= getHeight() && diffY > getHeight() / 2)
+				return getStyle().backgroundNext;
+			else if (diffY >= 0 && diffY <= getHeight() / 2) return getStyle().backgroundPrev;
+		}
+		return getStyle().background;
+	}
+
+	@Override
+	public void draw (Batch batch, float parentAlpha) {
+		currentBackground.draw(batch, getX(), getY(), getWidth(), getHeight());
+		// Center the text
+		layout = new GlyphLayout(style.font, model.getValue().toString());
+		style.font.draw(batch, model.getValue().toString() + "", getX() + style.background.getLeftWidth(),
+			getY() + (getHeight() + layout.height) / 2);
+	}
+
+	@Override
+	public float getPrefWidth () {
+		// Min of borders + text width or drawable width
+		layout = new GlyphLayout(style.font, model.getValue().toString());
+		return Float.max(style.background.getMinWidth(),
+			style.background.getLeftWidth() + layout.width + style.background.getRightWidth());
+	}
+
+	@Override
+	public float getPrefHeight () {
+		// Min of borders + text height or drawable height
+		layout = new GlyphLayout(style.font, model.getValue().toString());
+		return Float.max(style.background.getMinHeight(),
+			style.background.getTopHeight() + style.background.getBottomHeight() + layout.height);
+	}
+
+	public void setStyle (SpinnerStyle style) {
+		this.style = style;
+		invalidate();
+	}
+
+	public SpinnerStyle getStyle () {
+		return this.style;
+	}
+
+	public void setModel (SpinnerModel model) {
+		this.model = model;
+		invalidate();
+	}
+
+	public SpinnerModel getModel () {
+		return this.model;
+	}
+
+	/** This class defines the appearance of a Spinner. it includes the font used and the drawables for idle state, next button
+	 * pressed and previous button pressed. (See gdx-tests SpinnerTest) */
+	public static class SpinnerStyle {
+
+		public Drawable background, backgroundNext, backgroundPrev; // Backgrounds for the spinner states
```

Comment:
I am not sure if the way it is drawn (single background with buttons baked in) is the correct approach, but why not.
However it would be nice to also add on-hover images, to reach visual parity with libGDX's `Button`.